pyomo/contrib/mindtpy/util.py

```diff
@@ -932,3 +945,13 @@ def get_primal_integral(solve_data, config):
 
     config.logger.info(' {:<25}:   {:>7.4f} '.format('Primal integral', primal_integral))
     return primal_integral
+
+def load_solution_appsi(opt, config):
```

Comment:
This won't be neccesary once the APPSI issues are addressed, right?

Reply:
I would recommend something like the following here:

```
def load_solution_appsi(results: appsi.base.Results, config):
    assert isinstance(config.mip_solver, appsi.base.Solver)
    if results.best_feasible_objective is not None:
        results.solution_loader.load_vars()
```

Then there is no need to distinguish between specific solvers and no need to access private attributes.

Reply:
Great. Thanks.

Reply:
Oh, ignore that. This is using the wrapper to make the solver look like a standard solver. Instead, you should be able to do the following:

```
def load_solution_appsi(model, results):
    if pyo.check_optimal_termination(results):
        model.solutions.load_from(results)
```